src/main/java/clients/symphony/api/DatafeedClient.java

```diff
@@ -1,109 +1,50 @@
-                logger.info("Created new datafeed {} for bot {}", datafeedId.getId(),
-                    botClient.getBotUsername());
+import model.datafeed.DatafeedV2;
 
-                // if the reuseDatafeedID config isn't set (null), we assume its default value as true
-                if (botClient.getConfig().getReuseDatafeedID() == null || botClient.getConfig().getReuseDatafeedID()) {
-                    writeDatafeedIdToDisk(botClient.getConfig(), datafeedId.getId());
-                }
-
-                return datafeedId.getId();
-            }
-        }
-    }
-
-    private void writeDatafeedIdToDisk(SymConfig config, String datafeedId) {
-        String agentHostPort = config.getAgentHost() + ":" + config.getAgentPort();
-
-        try (FileWriter fw = new FileWriter(botClient.getDatafeedIdFile())) {
-            fw.write(datafeedId + "@" + agentHostPort);
-            fw.flush();
-        } catch (IOException ex) {
-            logger.error(ex.getMessage());
-        }
-    }
-
-    public List<DatafeedEvent> readDatafeed(String id) throws SymClientException {
-        WebTarget webTarget = botClient.getAgentClient().target(config.getAgentUrl());
-        Invocation.Builder builder = webTarget
-            .path(AgentConstants.READDATAFEED.replace("{id}", id))
-            .request(MediaType.APPLICATION_JSON)
-            .header("sessionToken", botClient.getSymAuth().getSessionToken())
-            .header("keyManagerToken", botClient.getSymAuth().getKmToken());
-
-        List<DatafeedEvent> datafeedEvents = null;
+import java.util.List;
 
-        logger.debug("Reading datafeed {}", id);
-        try (Response response = builder.get()) {
-            if (response.getStatusInfo().getFamily() != SUCCESSFUL) {
-                logger.error("Datafeed read error for request: {}", webTarget.getUri());
-                handleError(response, botClient);
-            } else if (response.getStatusInfo().getFamily() == CLIENT_ERROR) {
-                ((SymLoadBalancedConfig) config).rotateAgent();
-            } else {
-                if (response.getStatus() == NO_CONTENT.getStatusCode()) {
-                    datafeedEvents = new ArrayList<>();
-                } else {
-                    datafeedEvents = response.readEntity(DatafeedEventsList.class);
-                }
-            }
-            return datafeedEvents;
-        }
-    }
+/**
+ * DatafeedClient interface for handling datafeed endpoints.
+ */
+public interface DatafeedClient {
+
+    /**
+     * Create a new datafeed for bot.
+     *
+     * @return          created datafeed id
+     */
+    String createDatafeed();
+
+    /**
+     * Read all the events in datafeed.
+     *
+     * @param   id      id of the datafeed to be read
+     * @param   ackId   ackId given to delete all the events in datafeed
+     * @return          list of events in the datafeed
+     */
+    List<DatafeedEvent> readDatafeed(String id, String... ackId);
```

Comment:
aren't we breaking the compatibility here by changing it to an interface? (and by changing some of the methods)

Reply:
I don't understand your comment. As I thought, using interface help to avoid breaking the compatibility, isn't it?

Reply:
For instance let's say that a user of the SDK (an already released version) was overloading the class, now that it is an interface his code does not compile anymore when he updates the SDK. As a SDK or library user I'm not very happy if they make incompatible changes :)
But maybe we don't mind about compatibility, I'm not sure what is the current strategy here.

Reply:
I never noticed this example before. The idea behind the interface is keep two version of datafeed is aligned, and it break the compatibility in your example.
Maybe, we can schedule a quick call with @symphony-elias to see how we can modify the strategy here?

Reply:
Correction the class is final so my example is not really valid here.

There is the constructor that was public, so by replacing it with an interface we break source compatibility here, although using the constructor was probably not an intended usage of the API (I guess we expect users to use clients.SymBotClient#getDatafeedClient)

I'm not sure if replacing a class with an interface breaks binary compatibility but perhaps we don't guarantee it

Yes we can talk it further but it would be interesting the opinion of someone that worked on the SDK before and know the strategy about compatibility (@symphonydarlys in the absence of @symphony-thibault ?)